packages/contracts/contracts/MarketRegistry.sol

```diff
@@ -494,12 +508,14 @@ contract MarketRegistry is
         string calldata _metadataURI
     ) public ownsMarket(_marketId) {
         setMarketURI(_marketId, _metadataURI);
-        setPaymentCycleDuration(_marketId, _paymentCycleDuration);
+
         setPaymentDefaultDuration(_marketId, _paymentDefaultDuration);
         setBidExpirationTime(_marketId, _bidExpirationTime);
         setMarketFeePercent(_marketId, _feePercent);
         setLenderAttestationRequired(_marketId, _lenderAttestationRequired);
         setBorrowerAttestationRequired(_marketId, _borrowerAttestationRequired);
+        setMarketPaymentType(_marketId, _newPaymentType);
+        setPaymentCycle(_marketId, _paymentCycleType, _paymentCycleDuration);
```

Comment:
can we create an internal function that call all of these and use it for both here (updating market settings) and creating a new market

noticing that we are not emitting the event for setting attestation required when creating a new market - best if we just make them do the exact same thing

Reply:
I meant for all of these function actually - not just the these 2